[core-kit] net-dialup/ppp Autogening

org-tekeli-borisp opened the pull request.

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp-script Autogen net-dialup/ppp-scripts package Remove `ppp-scripts` from legacy gentoo staging lists and add as standalone package

`ppp-scripts` was removed from deprecated package lists and reintroduced as a standalone package with its own templates and metadata. This change improves modularity and simplifies maintenance of the `ppp-scripts` functionality.

(cherry picked from commit aab186a812ac76a91768ae98c0263c215a8c1f4b)
(cherry picked from commit b767d4e904f0d95b8035a008076bda0aab1072aa)

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp Add net-dialup/ppp with initial Ebuild template and configs

Introduce the PPP (Point-to-Point Protocol) package, including its ebuild template, autogen configurations, supporting files such as module aliases, and documentation. This addition re-enables the package in all relevant `packages.yaml` files for proper inclusion.

(cherry picked from commit 91b14068ddda349bf220c88eccea46842d3f5caf)
(cherry picked from commit 4c0cbe2915e83144d58100a03589c64d5a31269a)

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp Add net-dialup/ppp Update ppp dependency from libcrypt to libxcrypt

Replaces the deprecated virtual/libcrypt with sys-libs/libxcrypt in the ppp template. This ensures compatibility with modern systems and resolves potential dependency issues.

(cherry picked from commit 1bc378e3351b894c39b18b522ccea86d17856937)
(cherry picked from commit ebd244ba861ae8c6d7b0e105bcd56273de11f7c0)

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp  Add ipv6 USE flag to ppp template

Introduce an ipv6 USE flag to enable better control over IPv6 support in PPP. This provides users with more flexibility in configuring their builds.

(cherry picked from commit 949fd9e7cda31310bf9f8f895e6e630590bb8ef2)
(cherry picked from commit 912bc5c98837b33893218c7384dbf8e79ea2da4d)

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp Move net-dialup/ppp and ppp-scripts to 'mark' directory.

Reorganized files and templates from 'next' to 'mark' directory to align with updated project structure conventions. Renamed specific files to maintain consistency during the migration.

geaaru commented on core-kit/mark/net-dialup/ppp-scripts/autogen.yaml:
```diff
@@ -0,0 +1,10 @@
+        desc: Common set of scripts for various PPP implementations
+        homepage: https://gentoo.org/
+        license: GPL-2
+        src_uri: https://distfiles.gentoo.org/distfiles/32/ppp-scripts-0.tar.xz
```
Can we unpack this package directly in $FILESDIR and to maintain our version like I did for baselayout? Wdyt?

org-tekeli-borisp replied:
@geaaru, are you able to edit the PR and add your proposed changes?

geaaru replied:
Not, because this PR is not been created in this repo but in your fork.

org-tekeli-borisp replied:
could you point me to the place?

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp-script Add src_unpack function and include ppp-scripts archive

The src_unpack function now extracts the ppp-scripts archive during the build process, ensuring all necessary files are available. Removed `src_uri` from autogen.yaml as the archive is now included in the repository.

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp-script Add modular PPP connection handling scripts.

Introduce scripts for managing PPP connections, including hooks for IP setup and teardown (ip-up and ip-down). Each step is modularized for flexibility, handling WINS server updates, DNS configuration, and service activation/deactivation, with OpenRC support.

org-tekeli-borisp committed:
[core-kit] net-dialup/ppp-script Update homepage URL and version in ppp-scripts metadata

Updated the homepage to the correct GitHub repository URL for better accuracy. Incremented the version from "0" to "1" to reflect the changes.
